aten/src/ATen/core/dispatch/Dispatcher.h

```diff
@@ -670,7 +701,19 @@ C10_ALWAYS_INLINE_UNLESS_MOBILE Return Dispatcher::call(const TypedOperatorHandl
     return callWithDispatchKeySlowPath<Return, Args...>(op, *step_callbacks, dispatchKeySet, kernel, std::forward<Args>(args)...);
   }
 #endif  // PYTORCH_DISABLE_PER_OP_PROFILING
-  return kernel.template call<Return, Args...>(op, dispatchKeySet, std::forward<Args>(args)...);
+
+#ifdef FBCODE_CAFFE2
+  if(profilingOperatorEvents()) {
+    auto& schema = op.schema();
+    auto schema_ref = std::reference_wrapper<const FunctionSchema>(schema);
+    EventRAII ev(fireOpStartUSDT, fireOpEndUSDT, schema_ref);
+    return kernel.template call<Return, Args...>(op, dispatchKeySet, std::forward<Args>(args)...);
```

Comment:
How about moving FireOpRAII into the template itself and avoid exposing it globally and two unnecessary binds? (one can even use `https://en.cppreference.com/w/cpp/memory/unique_ptr` with custom deleter, though this would look a bit ugly)
```suggestion
    struct FireOpRAII {
       FireOpRAII(at::RecordFunction::schema_ref_t sr): sr_(sr) { 
           fireOpStartUSDT(sr);
        }
       ~FireOpRAII() { fireOpEndUSDT(sr_); }
       at::RecordFunction::schema_ref_t sr;
    } event(op.schema());
    return kernel.template call<Return, Args...>(op, dispatchKeySet, std::forward<Args>(args)...);
```